### ISSUE TITLE
[bug]: 3.0.0b5 Breaks on 2nd run

### ISSUE THREAD

**Issue by TijuanaKez** [invoke-ai/InvokeAI#3748]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

macOS

### GPU

mps

### VRAM

16GB

### What version did you experience this issue on?

3.0.0b5

### What happened?

Installer script downloads models and runs fine 1st go.
After quitting and trying to relaunch the webui, or run the model downloader script, it always fails with "UnpicklingError: invalid load key, '\x00'.
The only way to get the webui or any of the scrip options to work again is to delete the sd-1 and sd-2 folders and let it download the models again.
It will then only work until quitting, and the Unpicking error starts again.

### Screenshots

```
[2023-07-13 16:58:03,971]::[InvokeAI]::INFO --> scanning /Users/me/invokeai-3/models for new models
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/me/invokeai-3/.venv/bin/invokeai-model-install:8 in <module>                              │
│                                                                                                  │
│   5 from invokeai.frontend.install import invokeai_model_install                                 │
│   6 if __name__ == '__main__':                                                                   │
│   7 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 8 │   sys.exit(invokeai_model_install())                                                       │
│   9                                                                                              │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/frontend/install/model_install  │
│ .py:782 in main                                                                                  │
│                                                                                                  │
│   779 │   │   )                                                                                  │
│   780 │   │   from invokeai.frontend.install import invokeai_configure                           │
│   781 │   │                                                                                      │
│ ❱ 782 │   │   invokeai_configure()                                                               │
│   783 │   │   sys.exit(0)                                                                        │
│   784 │                                                                                          │
│   785 │   try:                                                                                   │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/backend/install/invokeai_confi  │
│ gure.py:837 in main                                                                              │
│                                                                                                  │
│   834 │   │   if not config.model_conf_path.exists():                                            │
│   835 │   │   │   initialize_rootdir(config.root_path, opt.yes_to_all)                           │
│   836 │   │                                                                                      │
│ ❱ 837 │   │   models_to_download = default_user_selections(opt)                                  │
│   838 │   │   new_init_file = config.root_path / 'invokeai.yaml'                                 │
│   839 │   │   if opt.yes_to_all:                                                                 │
│   840 │   │   │   write_default_options(opt, new_init_file)                                      │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/backend/install/invokeai_confi  │
│ gure.py:604 in default_user_selections                                                           │
│                                                                                                  │
│   601 │   return opts                                                                            │
│   602                                                                                            │
│   603 def default_user_selections(program_opts: Namespace) -> InstallSelections:                 │
│ ❱ 604 │   installer = ModelInstall(config)                                                       │
│   605 │   models = installer.all_models()                                                        │
│   606 │   return InstallSelections(                                                              │
│   607 │   │   install_models=[models[installer.default_model()].path or models[installer.defau   │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/backend/install/model_install_  │
│ backend.py:96 in __init__                                                                        │
│                                                                                                  │
│    93 │   │   │   │    model_manager: ModelManager = None,                                       │
│    94 │   │   │   │    access_token:str = None):                                                 │
│    95 │   │   self.config = config                                                               │
│ ❱  96 │   │   self.mgr = model_manager or ModelManager(config.model_conf_path)                   │
│    97 │   │   self.datasets = OmegaConf.load(Dataset_path)                                       │
│    98 │   │   self.prediction_helper = prediction_type_helper                                    │
│    99 │   │   self.access_token = access_token or HfFolder.get_token()                           │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/backend/model_management/model  │
│ _manager.py:352 in __init__                                                                      │
│                                                                                                  │
│   349 │   │   self.cache_keys = dict()                                                           │
│   350 │   │                                                                                      │
│   351 │   │   # add controlnet, lora and textual_inversion models from disk                      │
│ ❱ 352 │   │   self.scan_models_directory()                                                       │
│   353 │                                                                                          │
│   354 │   def model_exists(                                                                      │
│   355 │   │   self,                                                                              │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/backend/model_management/model  │
│ _manager.py:817 in scan_models_directory                                                         │
│                                                                                                  │
│   814 │   │   │   │   │   │   │   if model_path.is_relative_to(self.app_config.root_path):       │
│   815 │   │   │   │   │   │   │   │   model_path = model_path.relative_to(self.app_config.root   │
│   816 │   │   │   │   │   │   │   try:                                                           │
│ ❱ 817 │   │   │   │   │   │   │   │   model_config: ModelConfigBase = model_class.probe_config   │
│   818 │   │   │   │   │   │   │   │   self.models[model_key] = model_config                      │
│   819 │   │   │   │   │   │   │   │   new_models_found = True                                    │
│   820 │   │   │   │   │   │   │   except NotImplementedError as e:                               │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/backend/model_management/model  │
│ s/stable_diffusion.py:160 in probe_config                                                        │
│                                                                                                  │
│   157 │   │   │   │   ckpt_config["model"]["params"]["unet_config"]["params"]["in_channels"]     │
│   158 │   │   │                                                                                  │
│   159 │   │   │   else:                                                                          │
│ ❱ 160 │   │   │   │   checkpoint = read_checkpoint_meta(path)                                    │
│   161 │   │   │   │   checkpoint = checkpoint.get('state_dict', checkpoint)                      │
│   162 │   │   │   │   in_channels = checkpoint["model.diffusion_model.input_blocks.0.0.weight"   │
│   163                                                                                            │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/invokeai/backend/model_management/model  │
│ s/base.py:406 in read_checkpoint_meta                                                            │
│                                                                                                  │
│   403 │   │   │   scan_result = scan_file_path(path)                                             │
│   404 │   │   │   if scan_result.infected_files != 0:                                            │
│   405 │   │   │   │   raise Exception(f"The model file \"{path}\" is potentially infected by m   │
│ ❱ 406 │   │   checkpoint = torch.load(path, map_location=torch.device("meta"))                   │
│   407 │   return checkpoint                                                                      │
│   408                                                                                            │
│   409 import warnings                                                                            │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/torch/serialization.py:815 in load       │
│                                                                                                  │
│    812 │   │   │   │   return _legacy_load(opened_file, map_location, _weights_only_unpickler,   │
│    813 │   │   │   except RuntimeError as e:                                                     │
│    814 │   │   │   │   raise pickle.UnpicklingError(UNSAFE_MESSAGE + str(e)) from None           │
│ ❱  815 │   │   return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args  │
│    816                                                                                           │
│    817                                                                                           │
│    818 # Register pickling support for layout instances such as                                  │
│                                                                                                  │
│ /Users/me/invokeai-3/.venv/lib/python3.10/site-packages/torch/serialization.py:1033 in           │
│ _legacy_load                                                                                     │
│                                                                                                  │
│   1030 │   │   │   f"Received object of type \"{type(f)}\". Please update to Python 3.8.2 or ne  │
│   1031 │   │   │   "functionality.")                                                             │
│   1032 │                                                                                         │
│ ❱ 1033 │   magic_number = pickle_module.load(f, **pickle_load_args)                              │
│   1034 │   if magic_number != MAGIC_NUMBER:                                                      │
│   1035 │   │   raise RuntimeError("Invalid magic number; corrupt file?")                         │
│   1036 │   protocol_version = pickle_module.load(f, **pickle_load_args)                          │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
UnpicklingError: invalid load key, '\x00'.
```

**Comment by stofi** [comment]
@TijuanaKez 
Had the same problem after I manually deleted a model in the Finder. Fixed it by deleting `.DS_Store` files in the invokeai directory:

```bash
find . -name ".DS_Store" -delete
```

**Comment by TijuanaKez** [comment]
> @TijuanaKez Had the same problem after I manually deleted a model in the Finder. Fixed it by deleting `.DS_Store` files in the invokeai directory:
> 
> ```shell
> find . -name ".DS_Store" -delete
> ```

Ahh, knew it must be a hidden file. But I was attempting to find them with Show Hidden Files in Finder which still doesn't show .DS_Store files.

Added to invoke.sh for now as workaround, but it is obviously a genuine bug for macOS.

```
# Ensure we're in the correct folder in case user's CWD is somewhere else
scriptdir=$(dirname "$0")
cd "$scriptdir"

find ./models -name ".DS_Store" -delete

. .venv/bin/activate
```

Thanks.

**Comment by twopiearr** [comment]
Can confirm this is still happening in 3.0.1.

**Comment by dang-lbc** [comment]
Still a bug in InvokeAI 3.1.0.  Thank you all for figuring out what the cause of the problem is! I've been searching the forums for weeks to figure out why InvokeAI kept failing to launch every subsequent time, after a successful first launch, on my Mac. I followed lots of dead ends about clearing Python caches and whatnot.

Unfortunately, I could not get the workaround by @TijuanaKez to work for me. Does it only work if you set AppleShowAllFiles to TRUE in Terminal? I used an app called _Find Any File_ to root out the hidden .DS_Store files and delete them from my InvokeAI model folders. Then InvokeAI was able to launch again! Sooner or later, Finder will regenerate the .DS_Store files, but at least now I know a temporary fix. Thanks again! Hope the developers can fix this bug for Mac users.
